osl/sandbox.go

```diff
@@ -51,6 +51,10 @@ type Sandbox interface {
 	// RemoveAliasIP removes the passed IP address from the named interface
 	RemoveAliasIP(ifName string, ip *net.IPNet) error
 
+	// DisableARPForVIP disables ARP replies and requests for VIP addresses
+	// on a particular interface
+	DisableARPForVIP(ifName string) error
```

Comment:
better DisableARPForIfc?

Reply:
Hrmm.. Don't object to renaming, but the suggestion is less accurate.  The function doesn't disable ARP for all addresses.  It just disables it for addresses assigned to a different interface.  We really only care that the VIP addresses have ARP suppressed for purposes of this API.  But the operation *does* change behavior slightly.  Say you had:

```
                            +---------------+
       eth0(10.1.0.2) ------+   container   +------ eth1(10.2.0.7)
                            +---------------+
```

The way containers are currently configured, if you ARPed 10.2.0.7 on eth0, you'd get a response back indicating that that address was on that interface.  The ARP configurations that this function performs would prevent that.  (the contaienr would ignore incoming ARPs for 10.2.0.7 on eth0)  Ideally, we would be configuring this behavior ONLY for the VIP addresses (which get assigned to `lo`).  Unfortunately, there isn't a nice way to do that without using arptables as far as I know.   And requiring arptables and the like on docker installations is not feasible.

As far as ARPing across interfaces goes, my personal opinion is that this behavior should be the restriction that this function puts in place should be the default.  But the Linux devs disagree.  In any case I don't think that the more relaxed behavior is something an application developer should rely on.  (see https://lwn.net/Articles/45373/)  So the restriction *should* be ok.  It would also be a property of the network, in any case.  (i.e. only applies to endpoints that are on a "dsr"-option-enabled "overlay" network).

Reply:
I see the point, it's difficult to express with a concise name. nvm for the moment then